common/performance.py

```
@@ -0,0 +1,721 @@
+        # Initialize cache manager if needed
+        self._init_cache_manager()
+
+    def _init_cache_manager(self):
+        """Initialize cache manager for this view if not already registered."""
+        primary_model = getattr(self, "primary_model", None)
+        if primary_model:
+            manager_name = f"{primary_model.__name__}_view_cache"
+
+            if not cache_registry.get(manager_name):
+                # Create a generic cache manager for this view
+                def view_data_builder():
```

Ruff (0.12.2)

Missing return type annotation for private function `view_data_builder`

(ANN202)